### PR TITLE
Properly quote file names in listFiles query for GDrive

### LINKED ISSUE #13189: Single quotes not escaped when fetching files from mounted volume
I'm having trouble syncing files on my local drive when filenames contains a single quote and are issued from my GoogleDrive mounted volume. 

Windows client crash : 
11:52     Gdrive/Documents/StBruno - Oenologie/2014 - Novembre - Alsace/CREMANT D'ALSACE.pdfownCloud       Error downloading https://cloud.uldosphere.org/remote.php/webdav/Gdrive/Documents/StBruno - Oenologie/2014 - Novembre - Alsace/CREMANT D'ALSACE.pdf - server replied: Internal Server Error

Apache access.log ispretty obvious : 

cloud.uldosphere.org:443 localhost - uld [09/Jan/2015:11:55:19 +0100] "GET /remote.php/webdav/Gdrive/Documents/StBruno%20-%20Oenologie/2014%20-%20Novembre%20-%20Alsace/CREMANT%20D'ALSACE.pdf HTTP/1.1" 500 860 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"

Filename CREMANT%20D'ALSACE.pdf is not escaped on the single quote.

### PR DESCRIPTION
**Pull request by PVince81**
Fixes https://github.com/owncloud/core/issues/13189

### CODE REVIEW

**Comment by karlitschek** [comment]
nice :+1:

File: apps/files_external/lib/google.php [hunk 1]
```diff
@@ -103,7 +103,7 @@ private function getDriveFile($path) {
 				if (isset($this->driveFiles[$path])) {
 					$parentId = $this->driveFiles[$path]->getId();
 				} else {
-					$q = "title='".$name."' and '".$parentId."' in parents and trashed = false";
+					$q = "title='".rawurlencode($name)."' and '".$parentId."' in parents and trashed = false";
```
Comment (LukasReschke) [inline apps/files_external/lib/google.php]: Can you quote the parentId here as well even if unneeded as it always should be numeric? Just to be always on the safe side.

Comment (PVince81) [inline apps/files_external/lib/google.php]: Good point, will do

**Comment by PVince81** [comment]
I ran the unit tests and there were no new issues

**Comment by PVince81** [comment]
@LukasReschke fixed

**Comment by PVince81** [comment]
Damn, I pushed too quickly again and messed up Jenkins.
Will raise a Jenkins PR

**Comment by PVince81** [comment]
@LukasReschke please have another look

**Comment by PVince81** [comment]
- [ ] BUG: with spaces

**Comment by icewind1991** [comment]
code looks good :+1:

**Comment by LukasReschke** [comment]
@karlitschek This patch was not complete and thus labelled with `3 - To Develop` as it will likely cause problems with files containing space characters as stated in https://github.com/owncloud/core/issues/13189#issuecomment-86644750. Thus I reverted the merge for now.

**Comment by LukasReschke** [comment]
I will create a new pull request and take a look at the issue.

**Comment by LukasReschke** [comment]
https://github.com/owncloud/core/pull/15336

**Pull request merged**